bazel/external/lightstep.genrule_cmd

```diff
@@ -0,0 +1,20 @@
+# This is a Bazel genrule script. It acts like Bash, but has some extra
+# substitutions for locating dependencies.
+#
+# https://docs.bazel.build/versions/master/be/general.html#genrule
+
+CONFIGURE="$${PWD}/$(location :configure)"
```

Comment:
You might want to propagate `CC_FLAGS` and `CC` here too.
https://docs.bazel.build/versions/master/be/make-variables.html

Reply:
`CC` is `"/usr/bin/g++"` even when building with Clang, and `CC_FLAGS` is always empty: https://github.com/bazelbuild/bazel/issues/489

At this point I'm not certain that `--copt` actually gets propagated to genrule commands in any way. That might be WAI because they really don't want people building C++ in genrules, but it sure makes life hard.

Reply:
@jmillikin-stripe do you want something like `use_default_shell_env` that exists for rule contexts? https://docs.bazel.build/versions/master/skylark/lib/ctx.html

Reply:
Not sure -- does `use_default_shell_env` propagate the `--copt` flag through to an env var?

I think I've got something working based on `select()`, but it's not beautiful. It might be worth bringing this up internally to see if https://github.com/bazelbuild/bazel/issues/489 can be relaxed.

Reply:
Oh hey, I dug up an interesting bad thing. Bazel is being run without `--batch`, which means it caches the value of `$CC` between invocations (in a background process). That's why `CC` was always `g++`, it was using the older `$CC` from `setup_gcc_toolchain`.

Fixing this let Clang be used properly in the genrule, but caused more failures because the `./configure` helper scripts trigger Clang's leak detector.